src/lib/y2storage/planned/can_be_formatted.rb

```diff
@@ -98,14 +98,15 @@ def subvolumes?
         btrfs? && !subvolumes.nil? && !subvolumes.empty?
       end
 
-      # Removes from #subvolumes all the plannes subvolumes that would be
-      # shadowed by another device mounted in any of the given mount points
+      # Planned subvolumes (from #subvolumes) that would be shadowed by any of
+      # the given planned devices.
       #
-      # @param other_mount_points [Array<String>] mount points of the other
-      #   devices in the filesystem
-      def remove_shadowed_subvolumes!(other_mount_points)
-        return if subvolumes.empty?
-        self.subvolumes = subvolumes.reject { |subvol| subvol.shadowed?(other_mount_points) }
+      # @param all_devices [Array<Planned::Device>] all the devices planned for
+      #   the system.
+      def shadowed_subvolumes(all_devices)
+        other_devices = all_devices - [self]
```

Comment:
Because a given filesystem would always shadow all its subvolumes. I mean, imagine the filesystem `/home` including subvolumes for `/home/one` and `/home/two`.

In theory, the path of the filesystem shadows the path of its subvolumes... if they weren't be subvolumes of that particular fs.

Hmmm, not sure if that was a good explanation :-)